### PR TITLE
docs(config): moves config overview to a new section

### PR DESCRIPTION
**Pull request by PaulRMellor**
### Documentation change

Content from the Configuring guide is being moved to the Deploying guide.
The Configuring guide is being positioned as a pure schema reference guide.
Instructions to deploy, configure and administer Strimzi will sit together in the Deploying guide.

This PR moves content that provides an overview of configuring a deployment, including a description of the custom resources available and the example configuration files that can be used.

This PR also includes:

- Updates to the abstracts to the config sections
- restructuring of the deploying guide to position config related content togetehr in a single chapter, including use of ConfigMaps and configuration providers.

image

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]
2 nits, but GTM otherwise.

File: documentation/assemblies/configuring/assembly-config.adoc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+[id='overview-{context}']
+= Configuring a deployment
+
+[role="_abstract"]
+Configure and manage a Strimzi deployment to your precise needs using Strimzi custom resources.
+Strimzi provides example custom resources with each release, allowing you to configure and create instances of supported Kafka components.
+Fine-tune your deployment by configuring custom resources to include additional features according to your specific requirements. 
+For specific areas of configuration, namely metrics, logging, and external configuration for Kafka Connect connectors, you can also use `ConfigMap` resources.
+By using a `ConfigMap` resource to incorporate configuration, you centralize maintenance and enable reuse across different components.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/assemblies/configuring/assembly-config.adoc:13]: Removed the reuse bit

File: documentation/assemblies/configuring/assembly-config.adoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+[id='overview-{context}']
+= Configuring a deployment
+
+[role="_abstract"]
+Configure and manage a Strimzi deployment to your precise needs using Strimzi custom resources.
+Strimzi provides example custom resources with each release, allowing you to configure and create instances of supported Kafka components.
+Fine-tune your deployment by configuring custom resources to include additional features according to your specific requirements. 
+For specific areas of configuration, namely metrics, logging, and external configuration for Kafka Connect connectors, you can also use `ConfigMap` resources.
+By using a `ConfigMap` resource to incorporate configuration, you centralize maintenance and enable reuse across different components. 
+You can also use configuration providers to load configuration from external sources.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/assemblies/configuring/assembly-config.adoc:14]: Changed as follows:

> You can also use configuration providers to load configuration from external sources, which we recommend for supplying the credentials for Kafka Connect connector configuration.

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]
LGTM

**Pull request merged**